src/Angor/Shared/Services/RelayService.cs

```diff
@@ -96,65 +104,130 @@ public Task LookupSignaturesDirectMessagesForPubKeyAsync(string nostrPubKey, Dat
 
             var subscriptionKey = nostrPubKey + "DM";
 
+            // Process cached events first
+            var cachedEvents = _nostrEventCacheService.GetCachedEvents(subscriptionKey);
```

Comment:
I just wonder what efficiency gains do we get here? we anyway create a subscription and call the relay.
Events will be fired twice once from cache and then again from each relay.